apiary-receivers-parent/apiary-receiver-common/src/main/java/com/expedia/apiary/extensions/receiver/common/messaging/MessageDeserializer.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (C) 2018-2019 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.apiary.extensions.receiver.common.messaging;
+
+import com.expedia.apiary.extensions.receiver.common.error.ApiaryReceiverException;
+import com.expedia.apiary.extensions.receiver.common.event.ListenerEvent;
+
+public interface MessageDeserializer {
+
+  <T extends ListenerEvent> T unmarshal(String payload) throws ApiaryReceiverException;
```

Comment:
I'd expect the classes that deal with serialization and deserialization to throw `SerDeExceptions` since we have a more specific subclass of `ApiaryReceiverException` which I'd expect to be thrown more generally.

Reply:
Yep agreed, this is changed now, thanks!